src/redux/rockets/rocketSlice.js

```diff
@@ -0,0 +1,40 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+const initialState = {
+  rocketLists: [],
+  status: false,
+};
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  const response = await axios.get(url);
+  return response.data;
+});
+const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {
+    reserveRocket: (state, { payload }) => {
+      const rockets = state.rocketLists.map((rocket) => {
+        if (rocket.id === payload) return { ...rocket, reserved: !rocket.reserved };
+        return rocket;
+      });
+
+      return { ...state, rocketLists: rockets };
+    },
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchRockets.pending, (state) => ({ ...state, status: true }))
+      .addCase(fetchRockets.fulfilled, (state, action) => ({
+        ...state,
+        rocketLists: action.payload,
+      }));
```

Comment:
- [ ]  `optional` Kindly  add an extra reducer for rejected action as sometime due to client error or API error we could get rejected result so it will be helpful to track it down when our APPLICATION break